### PR TITLE
refactor(typing): reduce aws_lambda_powertools.shared.types usage 

### LINKED ISSUE #4897: Tech debt: Remove usages of aws_lambda_powertools.shared.types
### Why is this needed?

Remove usages of aws_lambda_powertools.shared.types imports and replace them by either types from the standard library that are already compatible with Python >= 3.8 or types from typing_extensions. Remove the aws_lambda_powertools.shared.types package too.

### Which area does this relate to?

Other

### Suggestion

_No response_

### Acknowledgment

- [X] This request meets [Powertools for AWS Lambda (Python) Tenets](https://docs.powertools.aws.dev/lambda/python/latest/#tenets)
- [ ] Should this be considered in other Powertools for AWS Lambda languages? i.e. [Java](https://github.com/aws-powertools/powertools-lambda-java/), [TypeScript](https://github.com/aws-powertools/powertools-lambda-typescript/), and [.NET](https://github.com/aws-powertools/powertools-lambda-dotnet/)

### PR DESCRIPTION
**Pull request by ericbn**
**Issue number:** #4897 

### Changes

Reduce `aws_lambda_powertools.shared.types` usage. The goal was to remove the package as a whole, but the custom `AnyCallableT` and `JSONType` defined there cannot be replaced by any type provided in typing or typing_extensions.

### User experience

As discussed in #4607, this is an intermediary step towards adding `from __future__ import annotations` to all files. The refactoring in this PR will help ruff better help us refactor the code once we introduce the TCH and UP rules as discussed in the main issue.

## Checklist

If your change doesn't seem to apply, please leave them unchecked.

* [X] [Meet tenets criteria](https://docs.powertools.aws.dev/lambda/python/#tenets)
* [X] I have performed a self-review of this change
* [X] Changes have been tested
* [ ] Changes are documented
* [X] PR title follows [conventional commit semantics](https://github.com/aws-powertools/powertools-lambda-python/blob/develop/.github/semantic.yml)

<details>
<summary>Is this a breaking change?</summary>

**RFC issue number**:

Checklist:

* [ ] Migration process documented
* [ ] Implement warnings (if it can live side by side)

</details>

## Acknowledgment

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

**Disclaimer**: We value your time and bandwidth. As such, any pull requests created on non-triaged issues might not be successful.

### CODE REVIEW

File: aws_lambda_powertools/shared/types.py [hunk 1]
```diff
@@ -1,25 +1,5 @@
...
-# So for python < 3.10 we use the backport from typing_extensions.
-if sys.version_info >= (3, 10):
-    from typing import TypeAlias, get_args, get_origin
-else:
-    from typing_extensions import TypeAlias, get_args, get_origin
+from typing import Any, Callable, Dict, List, TypeVar, Union
 
 AnyCallableT = TypeVar("AnyCallableT", bound=Callable[..., Any])  # noqa: VNE001
 # JSON primitives only, mypy doesn't support recursive tho
 JSONType = Union[str, int, float, bool, None, Dict[str, Any], List[Any]]
```
Comment (leandrodamascena) [inline aws_lambda_powertools/shared/types.py]: We only use `JSONType` in the Feature Flags utility and a test, so what do you think about moving it there?

**Comment by leandrodamascena** [comment]
Reviewing now.

**Review by leandrodamascena — CHANGES_REQUESTED** [review]
Hey @ericbn I did some changes here.

-  I fixed the documentation, some highlights were wrong because we changed imports.
- I removed JSONType from shared and moved it to Feature Flags, makes more sense. 
- I fixed absolute imports in Feature Flags - unrelated, but always good.

Can you please take a look before we approve and merge?

File: aws_lambda_powertools/shared/types.py (hunk 1 above)
Comment (leandrodamascena) [reply aws_lambda_powertools/shared/types.py]: It's solved.

**Comment by ericbn** [comment]
Oi @leandrodamascena. All looking good to me. Good catch on the absolute imports and good to know about `hl_lines` which I totally missed.

**Review by leandrodamascena — APPROVED** [review]
APPROVED! @ericbn 🥇

**Comment by leandrodamascena** [comment]
> Oi @leandrodamascena.

Olá :D

**Pull request merged**